Applications/Spire/Include/Spire/Ui/ColorBox.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef SPIRE_COLOR_BOX_HPP
+#define SPIRE_COLOR_BOX_HPP
+#include <memory>
+#include <QColor>
+#include <QWidget>
+#include "Spire/Spire/LocalValueModel.hpp"
+#include "Spire/Ui/FocusObserver.hpp"
+#include "Spire/Ui/PressObserver.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /** A ValueModel over a QColor. */
+  using ColorModel = ValueModel<QColor>;
+
+  /** A ValueModel over an optional QColor. */
+  using OptionalColorModel = ValueModel<boost::optional<QColor>>;
```

Comment:
What is the use case for the ```OptionalColorModel```? It doesn't look like it's used anywhere and I believe ```QColor``` already represents its own invalid value though ```QColor::isValid()```:

https://doc.qt.io/qt-6/qcolor.html#isValid